api/spec/requests/spree/api/promotion_application_spec.rb

```diff
@@ -20,6 +20,7 @@ module Spree::Api
 
       it "can apply a coupon code to the order" do
         expect(order.total).to eq(110.00)
+        expect(Spree::Deprecation).to receive(:warn)
```

Comment:
I'm not sure to understand why specs in these files are needed, maybe we can move them into `api/spec/requests/spree/api/coupon_codes_controller_spec.rb` ? Of course we should also remove the `expect(Spree::Deprecation).to receive(:warn)` and make them use the new endpoint. WDYT?

Reply:
Sounds good to me! Although I'm assuming we'd need another PR to remove duplication, am I right? Like in [here](https://github.com/nebulab/solidus/pull/19#discussion_r228501278)

Reply:
Makes perfect sense!